utils/messages.js

```diff
@@ -185,15 +195,54 @@ function parseMessageFile(fileName, details, type, callback) {
     }
     else {
       extractFields(
-        content, details, type, function(error, constants, fields) {
+        content, details, type, function(error, aggregate) {
           if (error) {
             callback(error);
-          }
-          else {
-            details.constants = constants;
-            details.fields    = fields;
-            details.md5       = calculateMD5(details);
-            callback(null, details);
+          } else {
+            if (type == "msg") {
+              details.constants = aggregate[0].constants;
+              details.fields    = aggregate[0].fields;
+              details.md5       = calculateMD5(details, "msg");
+              callback(null, details);
+            } else if (type == "srv") {
+              // services combine the two message types to compute the
+              // md5sum
+              var rtv = {
+                // we need to clone what's already there in details
+                // into the sub-objects
+                request: JSON.parse(JSON.stringify(details)),
+                response: JSON.parse(JSON.stringify(details))
+              };
+              rtv.request.constants = aggregate[0].constants;
+              rtv.request.fields = aggregate[0].fields;
+              rtv.response.constants = aggregate[1].constants;
+              rtv.response.fields = aggregate[1].fields;
+              rtv.request.md5 = rtv.response.md5 = calculateMD5(rtv, "srv");
```

Comment:
That should be fine. I would prefer having a proper service class as well, rather than using the Message class for everything. But I believe we'll need to coordinate this change, for our two class definitions in order to be able to also update where the md5 is being used, such as in ServiceClient: `this._messageHandler.Response.md5sum()`  would need to change to `this._messageHandler.md5sum()`. Right?